cmd/thermal-recorder/snapshot.go

```diff
@@ -57,3 +60,42 @@ func newSnapshot(lastFrame int) (*cptvframe.Frame, error) {
 	}
 	return f, nil
 }
+
+func newSnapshotRecording() error {
+	mu.Lock()
+	defer mu.Unlock()
+
+	if processor == nil {
+		return errors.New("reading from camera has not started yet")
+	}
+
+	processor.StartSnapshot = true
+	return nil
+}
+
+func snapshotRecordingTriggers(processor *motion.MotionProcessor, window window.Window) {
+	if window.NoWindow {
+		log.Println("no recording window so will make snapshot every 12 hours")
+		triggerTime := time.Now().Add(time.Minute)
+		for {
+			time.Sleep(time.Until(triggerTime))
+			log.Println("making snapshot")
+			processor.StartSnapshot = true
+			triggerTime = triggerTime.Add(time.Hour * 12)
+		}
+	}
+
+	// Make snapshot at start of window.
+	sleepTime := time.Minute
+	if !window.Active() {
+		sleepTime = time.Until(window.NextStart()) + time.Minute
+	}
+	time.Sleep(sleepTime)
```

Comment:
Looks like we will get  a snapshot everytime the camera connects?

Reply:
Yeah true. I have updated it so it will only make a recording if it is in the first 5 minutes of the recording window.